### DISCUSSION TITLE
Allow workflow dependencies on other workflows

### DISCUSSION THREAD

**Discussion by nihonjinrxs** [community/community#12688] [Actions]
It would be helpful to have a `needs` key like we have for a job, but for workflows, that would allow specifying another workflow's completion prior to running it.

We have a monorepo setting where many of our projects only get things run for them when it's relevant. In that setting, we often have situations where tests run for everything that has changed from the base ref, but different deployment workflows need to be executed for different projects. As is, we need to define multiple workflows with either a single composite action for the test workflows or all the relevant testing steps, duplicated in each workflow alongside the appropriate deployment steps. This results in lumping all the step logs and success information about testing, linting, etc., together in the same job as the deploy.

This would be a whole lot cleaner and more understandable if we could define our testing and linting as separate workflows, then define each deploy workflow with appropriate conditions and a simple key at the top referencing workflow names:
```{yml}
  needs: [ 'tests', 'e2e-backend', 'e2e-frontend', 'lint' ]
```
or workflow file paths:
```{yml}
  needs:
    - '.github/workflows/tests'
    - '.github/workflows/e2e-backend'
    - '.github/workflows/e2e-frontend'
    - '.github/workflows/lint'
```

Is there some way to do this I'm not seeing? If not, is there a possibility of adding this enhancement to workflows?

**Comment by ethomson** [comment]
The way that we'd recommend doing this today is using composites - which it sounds like you're doing. It sounds like you want more visibility into the execution of those composite jobs - you're thinking of the tests as a separate thing which should fail in this case. I think that makes sense, and I'll talk with the team about this and give some thoughts to ways that we might be able to expose it. I appreciate the `needs` suggestion, this seems like a possible way that we can give that visibility.

Thanks much for the feedback.
